pydantic/config.py

```diff
@@ -133,6 +133,8 @@ class User(BaseModel):
     Whether models are faux-immutable, i.e. whether `__setattr__` is allowed, and also generates
     a `__hash__()` method for the model. This makes instances of the model potentially hashable if all the
     attributes are hashable. Defaults to `False`.
+    It's not recommended to define both `frozen` in the `dataclass` decorator and `ConfigDict`.
+    If both are defined, the setting from the dataclass decorator will take priority.
```

Comment:
I would move this in the `dataclass` decorator docstring, `ConfigDict` shouldn't know about Pydantic dataclasses imo

Reply:
The current `dataclass` docstring is:

```
frozen: Determines if the generated class should be a 'frozen' `dataclass`, which does not allow its
    attributes to be modified after it has been initialized.
```

which I would rephrase as:

```
frozen: Determines if the generated class should be a 'frozen' `dataclass`, which does not allow its
    attributes to be modified after it has been initialized. If not set, the value from the provided `config` argument will be used (and will default to `False` otherwise).
```